query-planner-js/src/composedSchema/buildComposedSchema.ts

```diff
@@ -130,15 +130,24 @@ export function buildComposedSchema(document: DocumentNode): GraphQLSchema {
       type.astNode,
     );
 
-    const typeMetadata: FederationTypeMetadata = ownerDirectiveArgs
-      ? {
-          graphName: graphMap[ownerDirectiveArgs?.['graph']].name,
-          keys: new MultiMap(),
-          isValueType: false,
-        }
-      : {
-          isValueType: true,
-        };
+    let typeMetadata: FederationTypeMetadata;
+    if (ownerDirectiveArgs) {
+      const graph = graphMap.get(ownerDirectiveArgs.graph);
+      assert(
+        graph,
+        `@${ownerDirective.name} directive requires a \`graph\` argument`,
```

Comment:
Is this error message right? The assertion is checking that the graph argument names a known graph, not that the graph argument is provided. (Assuming that `@join__owner` is declared correctly, then yes, `graph` will need to be an enum and all enums are supposed to contribute to `graphMap`, but that's (I think) an unchecked assumption, and in any case if you're making that assumption then you don't need any error-checking here?) Perhaps split into two assertions?

Reply:
Yeah, the semantics are off on this one. I think I was assuming if it came back empty that's because `ownerDirectiveArgs.graph` was `undefined` but no need to be roundabout in the semantics there.

Hows: 385dc021 ?